### PR TITLE
migrating sonarqube connector to ballerina-1.0

### PR DESCRIPTION
**Pull request by sameeragunarathne**

### CODE REVIEW

File: src/sonarqube/Module.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Hello World
```
Comment (pramodya1994) [inline src/sonarqube/Module.md]: Should edit this

File: Ballerina.lock [hunk 2]
```diff
@@ -0,0 +1,4 @@
+org_name = "wso2"
+version = "0.8.21"
+lockfile_version = "1.0.0"
+ballerina_version = "1.0.0-rc1-SNAPSHOT"
```
Comment (Maninda) [inline Ballerina.lock]:
```suggestion
ballerina_version = "1.0.0"
```

File: src/sonarqube/sonarqube_utils.bal [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+# + return - Prepared `Error` instance
+public function prepareError(string message, error? err = ()) returns Error {
+    Error authError;
+    if (err is error) {
+        authError = error(AUTH_ERROR, message = message, cause = err);
+    } else {
+        authError = error(AUTH_ERROR, message = message);
+    }
+    return authError;
+}
```
Comment (Maninda) [inline src/sonarqube/sonarqube_utils.bal]:
```suggestion
}

```
Add a new line

File: src/sonarqube/sonarqube_endpoint.bal [hunk 4]
```diff
@@ -0,0 +1,760 @@
...
+# + return - If the execution is successful returns number of duplicated code blocks else an error
+public remote function getDuplicatedCodeBlocksCount(string projectKey) returns @tainted int|error {
+    var result = self.getMeasure(projectKey, DUPLICATED_BLOCKS_COUNT);
+    if (result is string) {
+        var intValue = ints:fromString(result);
+        if (intValue is int) {
+            return intValue;
+        } else {
+            error err = error(SONARQUBE_ERROR_CODE, message = "Error occurred when converting the given value to int");
+            panic err;
```
Comment (Maninda) [inline src/sonarqube/sonarqube_endpoint.bal]:
```suggestion
            check err;
```
Should catch the error and handle it.

Comment (Maninda) [reply src/sonarqube/sonarqube_endpoint.bal]: As this is not critical let's fix these issues later.

**Review by Maninda — APPROVED** [review]

**Review by Asitha — APPROVED** [review]

**Review by Asitha — APPROVED** [review]

**Pull request merged**